### PR TITLE
fix analytics with new format

### PR DESCRIPTION
**Pull request by AntoineJac**
### Summary

This PR is a fixed of https://github.com/Kong/kong/pull/12583
We need to refactor as many observability providers do not simplify ingest array. Here is a new nested format:

Current model is:
```json
  "ai": {
    "mistral": {
      "request_total_tokens": 648,
      "number_of_instances": 1,
      "instances": [
        {
          "meta": {
            "request_model": "mistral-tiny",
            "plugin_id": "1c783122-0223-4a03-9f1d-c71d2baab01e",
            "response_model": "mistral-tiny",
            "provider_name": "mistral"
          },
          "usage": {
            "prompt_token": 392,
            "total_tokens": 648,
            "completion_token": 256
          }
        }
      ],
      "request_prompt_tokens": 392,
      "request_completion_tokens": 256
    },
    "azure": {
      "request_total_tokens": 145,
      "number_of_instances": 1,
      "instances": [
        {
          "meta": {
            "request_model": "gpt-35-turbo",
            "plugin_id": "5df193be-47a3-4f1b-8c37-37e31af0568b",
            "response_model": "gpt-35-turbo",
            "provider_name": "azure"
          },
          "usage": {
            "prompt_token": 89,
            "total_tokens": 145,
            "completion_token": 56
          }
        }
      ],
      "request_prompt_tokens": 89,
      "request_completion_tokens": 56
    },
    "cohere": {
      "request_total_tokens": 149,
      "number_of_instances": 1,
      "instances": [
        {
          "meta": {
            "request_model": "command",
            "plugin_id": "30a63ab9-03c6-447a-9b25-195eb0acaeb2",
            "response_model": "command",
            "provider_name": "cohere"
          },
          "usage": {
            "prompt_token": 78,
            "total_tokens": 149,
            "completion_token": 71
          }
        }
      ],
      "request_prompt_tokens": 78,
      "request_completion_tokens": 71
    }
  }
```

New model if using several plugins:

```json
  "ai": {
    "ai-request-transformer": {
      "meta": {
        "request_model": "gpt-35-turbo",
        "provider_name": "azure",
        "response_model": "gpt-35-turbo",
        "plugin_id": "5df193be-47a3-4f1b-8c37-37e31af0568b"
      },
      "payload": {},
      "usage": {
        "prompt_token": 89,
        "completion_token": 56,
        "total_tokens": 145
      }
    },
    "ai-response-transformer": {
      "meta": {
        "request_model": "mistral-tiny",
        "provider_name": "mistral",
        "response_model": "mistral-tiny",
        "plugin_id": "1c783122-0223-4a03-9f1d-c71d2baab01e"
      },
      "payload": {},
      "usage": {
        "prompt_token": 168,
        "completion_token": 180,
        "total_tokens": 348
      }
    },
    "ai-proxy": {
      "meta": {
        "request_model": "gpt-35-turbo",
        "provider_name": "azure",
        "response_model": "gpt-35-turbo",
        "plugin_id": "546c3856-24b3-469a-bd6c-f6083babd2cd"
      },
      "payload": {},
      "usage": {
        "prompt_token": 28,
        "completion_token": 14,
        "total_tokens": 42
      }
    }
  },
```

### Checklist

- [ ] The Pull Request has tests
- [ ] A changelog file has been created under `changelog/unreleased/kong` or `skip-changelog` label added on PR if changelog is unnecessary. [README.md](https://github.com/Kong/gateway-changelog/README.md)
- [ ] There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE

### Issue reference

Fix #_[issue number]_

KAG-3759
KAG-4124

EDIT: format JSON snippets

### CODE REVIEW

File: kong/llm/drivers/shared.lua [hunk 1]
```diff
@@ -301,77 +276,62 @@ function _M.post_request(conf, response_object)
   -- analytics and logging
   if conf.logging and conf.logging.log_statistics then
     local provider_name = conf.model.provider
+    local plugin_name = conf.__key__:match('plugins:(.-):')
```
Comment (gruceo) [inline kong/llm/drivers/shared.lua:279]: what if there are no matches? can you add some error handling here? some tests too?

File: spec/03-plugins/38-ai-proxy/02-openai_integration_spec.lua [hunk 2]
```diff
@@ -35,26 +35,19 @@ local function wait_for_json_log_entry(FILE_LOG_PATH)
...
+      plugin_id = '6e7c40f6-ce96-48e4-a366-d109c169e444',
+      provider_name = 'openai',
+      request_model = 'gpt-3.5-turbo',
+      response_model = 'gpt-3.5-turbo-0613',
+    },
+    payload = {},
+    usage = {
+      completion_token = 12,
+      prompt_token = 25,
+      total_tokens = 37,
```
Comment (gruceo) [inline spec/03-plugins/38-ai-proxy/02-openai_integration_spec.lua:49]: perhaps update this test to use several plugins? or add a new test for several plugins? e.g. ai-proxy and the transformers plugins

Comment (AntoineJac) [reply spec/03-plugins/38-ai-proxy/02-openai_integration_spec.lua:49]: I added test for all AI plugins that will send statistic. Thanks

**Comment by tysoekong** [comment]
I can handle the Kong-EE cherry-picking and stuff, I have been set on to this to get everything done before next week.

Just need this merging to allow proceeding the other remaining PRs.

**Comment by tysoekong** [comment]
Set rebase when merging master back in! My fix conflicts now...

**Comment by tysoekong** [comment]
Tests should pass now

**Comment by tysoekong** [comment]
@gruceo passing now!

**Comment by tysoekong** [comment]
Wait I have pulled in commits now in this PR

**Comment by ttyS0e** [comment]
@gruceo ready!

**Review by gruceo — APPROVED** [review]
LGTM, wait for @brentos to approve then we can merge.

**Comment by gruceo** [comment]
@tysoekong @ttyS0e there are 14 commits in this PR, so we should squash.

**Comment by tysoekong** [comment]
@brentos Do you approve also?

**Review by brentos — APPROVED** [review]
LGTM!

**Comment by tysoekong** [comment]
@gruceo  Will it let you push squash and merge?

**Pull request merged**